src/inference/gibbs_conditional.jl

```diff
@@ -0,0 +1,134 @@
+"""
+    GibbsConditional(sym, conditional)
+
+A "pseudo-sampler" to manually provide analytical Gibbs conditionals to `Gibbs`.
+`GibbsConditional(:x, cond)` will sample the variable `x` according to the conditional `cond`, which
+must therefore be a function from a `NamedTuple` of the conditioned variables to a `Distribution`.
+
+# Examples
+
+```julia
+α_0 = 2.0
+θ_0 = inv(3.0)
+
+x = [1.5, 2.0]
+
+function gdemo_statistics(x)
+    # The conditionals and posterior can be formulated in terms of the following statistics:
+    N = length(x) # number of samples
+    x_bar = mean(x) # sample mean
+    s2 = var(x; mean=x_bar, corrected=false) # sample variance
+    return N, x_bar, s2
+end
+
+function gdemo_cond_m(c)
+    N, x_bar, s2 = gdemo_statistics(x)
+    m_n = N * x_bar / (N + 1)
+    λ_n = c.λ * (N + 1)
+    σ_n = sqrt(1 / λ_n)
+    return Normal(m_n, σ_n)
+end
+
+function gdemo_cond_λ(c)
+    N, x_bar, s2 = gdemo_statistics(x)
+    α_n = α_0 + (N - 1) / 2 + 1
+    β_n = s2 * N / 2 + c.m^2 / 2 + inv(θ_0)
+    return Gamma(α_n, inv(β_n))
+end
+
+@model gdemo(x) = begin
+    λ ~ Gamma(α_0, θ_0)
+    m ~ Normal(0, √(1 / λ))
+    x .~ Normal(m, √(1 / λ))
+end
+
+m = gdemo(x)
+
+sample(m, Gibbs(GibbsConditional(:λ, gdemo_cond_λ), GibbsConditional(:m, gdemo_cond_m)), 10)
+```
+"""
+struct GibbsConditional{S, C}
+    conditional::C
+
+    function GibbsConditional(sym::Symbol, conditional::C) where {C}
+        return new{sym, C}(conditional)
+    end
+end
+
+DynamicPPL.getspace(::GibbsConditional{S}) where {S} = (S,)
+alg_str(::GibbsConditional) = "GibbsConditional"
+isgibbscomponent(::GibbsConditional) = true
```

Comment:
I guess this should maybe also be part of another PR (since `GibbsComponent` is not removed yet)?
```suggestion
```